src/applet.rs

```diff
@@ -598,54 +946,102 @@ impl Application for SystemMonitorApplet {
+        let mut should_gpu = false;
+        let mut should_thermal = false;
+
+        for chart_config in &self.config.components {
+            match chart_config {
+                ComponentConfig::Cpu(_) => should_cpu = true,
+                ComponentConfig::Mem(_) => should_mem = true,
+                ComponentConfig::Net(_) => should_net = true,
+                ComponentConfig::Disk(_) => should_disk = true,
+                ComponentConfig::Gpu(_) => should_gpu = true,
+                ComponentConfig::Thermal(_) => should_thermal = true,
+            }
+        }
+
+        let mut subscriptions = vec![config_subscription()];
+
+        if should_cpu {
+            subscriptions.push(
+                cosmic::iced::time::every(Duration::from_millis(
+                    self.config.sampling.cpu.update_interval,
+                ))
+                .map(|_| Message::TickCpu),
+            );
         }
 
-        subs.push(config_subscription());
+        if should_mem {
+            subscriptions.push(
+                cosmic::iced::time::every(Duration::from_millis(
+                    self.config.sampling.mem.update_interval,
+                ))
+                .map(|_| Message::TickMem),
+            );
+        }
+
+        if should_net {
+            subscriptions.push(
+                cosmic::iced::time::every(Duration::from_millis(
+                    self.config.sampling.net.update_interval,
+                ))
+                .map(|_| Message::TickNet),
+            );
+        }
+
+        if should_disk {
+            subscriptions.push(
+                cosmic::iced::time::every(Duration::from_millis(
+                    self.config.sampling.disk.update_interval,
+                ))
+                .map(|_| Message::TickDisk),
+            );
+        }
+
+        if should_gpu {
+            subscriptions.push(
+                cosmic::iced::time::every(Duration::from_millis(
+                    self.config.sampling.gpu.update_interval,
+                ))
+                .map(|_| Message::TickGpu),
+            );
+        }
+
+        if should_thermal {
+            subscriptions.push(
+                cosmic::iced::time::every(Duration::from_millis(
+                    self.config.sampling.thermal.update_interval,
+                ))
+                .map(|_| Message::TickThermal),
+            );
+        }
 
-        Subscription::batch(subs)
+        cosmic::iced::Subscription::batch(subscriptions)
     }
 
     fn style(&self) -> Option<cosmic::iced_runtime::Appearance> {
         Some(cosmic::applet::style())
     }
 }
 
-pub fn base_background(theme: &Theme) -> container::Style {
+/// Changed to use `None` for background to make applet components transparent by default.
+pub fn base_background(_theme: &Theme) -> container::Style {
     container::Style {
-        background: Some(cosmic::iced::Color::from(theme.cosmic().primary.base).into()),
+        background: None,
```

Comment:
Wouldn't transparent by default makes it so we can't see the border of the charts?